ibis/backends/druid/__init__.py

```diff
@@ -0,0 +1,114 @@
+
+import ibis.backends.druid.datatypes  # noqa: F401
+import ibis.expr.datatypes as dt
+from ibis.backends.base.sql.alchemy import BaseAlchemyBackend
+from ibis.backends.druid.compiler import DruidCompiler
+
+
+class Backend(BaseAlchemyBackend):
+    name = 'druid'
+    compiler = DruidCompiler
+
+    def do_connect(
+        self,
+        host: str = "localhost",
+        port: int = 8082,
+        database: str | None = "druid/v2/sql",
+        **_: Any,
+    ) -> None:
+        """Create an Ibis client using the passed connection parameters.
+
+        Parameters
+        ----------
+        host
+            Hostname
+        port
+            Port
+        database
+            Database to connect to
+        """
+        url = sa.engine.url.make_url(f"druid://{host}:{port}/{database}?header=true")
+
+        self.database_name = "default"  # not sure what should go here
+
+        engine = sa.create_engine(url)
+
+        super().do_connect(engine)
+
+    @contextlib.contextmanager
+    def _safe_raw_sql(self, query, *args, **kwargs):
+        if not isinstance(query, str):
+            query = str(
+                query.compile(
+                    dialect=DruidDialect(), compile_kwargs=dict(literal_binds=True)
+                )
+            )
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                message="Dialect druid:rest will not make use of SQL compilation caching",
+                category=sa.exc.SAWarning,
+            )
+            with self.begin() as con:
+                yield con.exec_driver_sql(query, *args, **kwargs)
+
+    def _metadata(self, query: str) -> Iterable[tuple[str, dt.DataType]]:
+        raise NotImplementedError()
+
+    def _get_temp_view_definition(
+        self, name: str, definition: sa.sql.compiler.Compiled
+    ) -> str:
+        raise NotImplementedError()
+
+    def _has_table(self, table: str) -> bool:
+        query = sa.text(
+            """\
+SELECT COUNT(*) > 0 as c
+FROM INFORMATION_SCHEMA.TABLES
+WHERE TABLE_NAME = :table"""
+        ).bindparams(table=table)
+
+        with self._safe_raw_sql(query) as result:
+            return bool(result.scalar())
+
+    def _get_sqla_table(
+        self, name: str, schema: str | None = None, autoload: bool = True, **_: Any
+    ) -> sa.Table:
+        # If the underlying table (or more likely, view) has changed, remove it
+        # to ensure a correct reflection
+        if autoload and self._has_table(name):
+            self.meta.remove(sa.Table(name, self.meta))
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                message="|".join(
+                    (
+                        "Did not recognize type",
+                        "Dialect druid:rest will not make use of SQL compilation caching",
+                    )
+                ),
+                category=sa.exc.SAWarning,
+            )
```

Comment:
we could put a `super` call inside the `catch_warnings` context manager and keep the table selection business in `sql/alchemy/__init__.py`

Reply:
Sweet, thank you. That is much better!